### PR TITLE
[cordova][XWALK-2941] Update pack.py and pack_cordova_sample.py to support shared mode

### PR DESCRIPTION
**Pull request by zhuyongyong**
- Check pack.py in usecase-cordova-android-tests, cordova-sampleapp-android-tests successfully
- Check pack_cordova_sample.py with mobilespec, helloworld, remotedebugging and gallery successfully

### CODE REVIEW

File: tools/build/pack.py [hunk 1]
```diff
@@ -608,8 +608,12 @@ def packCordova(build_json=None, app_src=None, app_dest=None, app_name=None):
```
Comment (Honry) [inline tools/build/pack.py]: Currently Cordova 3.6 does not support arch in pack.py, please provide hint if people use -a when pack cordova 3.6 package.

File: tools/build/pack_cordova_sample.py [hunk 2]
```diff
@@ -323,7 +323,10 @@ def packSampleApp(app_name=None):
 
     orig_dir = os.getcwd()
```
Comment (Honry) [inline tools/build/pack_cordova_sample.py]: Ditto. 
And please also update the usage in README

**Comment by Honry** [comment]
Please provide your dry result.

File: tools/build/pack_cordova_sample.py [hunk 3]
```diff
@@ -441,6 +448,16 @@ def main():
             dest="bversion",
             action="store_true",
             help="show this tool's version")
+        opts_parser.add_option(
+            "-m",
+            "--mode",
```
Comment (Honry) [inline tools/build/pack_cordova_sample.py]: You didn't limit the value of package mode and package arch.

**Comment by Honry** [comment]
Please also provide the dry run result of all-in-one package:)

**Comment by zhuyongyong** [comment]
The cases of all-in-one package can run successfully.

**Comment by Honry** [comment]
LGTM

**Pull request merged**